### PR TITLE
Adding xcconfig for dynamic targets and fixing header and framework paths

### PR DESCRIPTION
**Pull request by bhariharan**
Done with ```SalesforceSDKCore``` for now, so that it can be reviewed. I'll be doing the same for the other frameworks.

### CODE REVIEW

**Comment by wmathurin** [comment]
Looks fine

**Comment by bhariharan** [comment]
Only outstanding piece is to update the path used for umbrella header generation.

File: configuration/Packaging-Dynamic.xcconfig [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//
+//  Packaging-Dynamic.xcconfig
+//  
+//
+//  Created by Bharath Hariharan on 12/11/15.
+//
```
Comment (wmathurin) [inline configuration/Packaging-Dynamic.xcconfig]: Copyright

**Comment by wmathurin** [comment]
:+1:

**Comment by bhariharan** [comment]
Fixed umbrella header generation run step path [here](https://github.com/bhariharan/SalesforceMobileSDK-iOS/commit/94565a4e0edf7d430d60251cdcca882e6e7b7038). Running validation tests now.

**Comment by jeanbovet** [comment]
@bhariharan looks good. Are you also changing the PUBLIC_HEADERS_FOLDER_PATH property of the target itself?

**Comment by bhariharan** [comment]
@jeanbovet I have already updated ```PUBLIC_HEADERS_FOLDER_PATH``` to ```$(CONTENTS_FOLDER_PATH)/Headers```, like the other dynamic framework targets I saw. This puts the headers in ```SalesforceSDKCore.framework/Headers```, for instance. Did I miss some targets?

**Comment by jeanbovet** [comment]
@bhariharan Ah, that's perfect, sorry, missed it!

**Comment by bhariharan** [comment]
<b>Note:</b>
* Any library or app that depends on ```SmartStore``` at some level should have ```CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES``` set to ```YES``` in its project file configuration.

**Pull request merged**